### PR TITLE
Refactor share_buttons view to remove screen reader duplication

### PR DESCRIPTION
**Pull request by kelbyhawn**
Updated the share buttons in `pegasus/sites.v3/code.org/views/share_buttons.haml` to use new link tag button classes (`%a.link-button…`) instead of a `%button` tag within an `%a` tag, which causes screen readers to re-read duplicate link tag and button tag labels. 
- Examples below are shown on https://code.org/promote
- The `share_buttons.haml` view is used elsewhere on the site and on [video lightboxes](https://github.com/code-dot-org/code-dot-org/blob/5d2c6486439f3cb8b24bbcef8c7963c24653b553/pegasus/sites.v3/code.org/views/index_video_reveal.haml#L22) so those have also been updated
- Deprecated unused share buttons styles

**Related PR:**
- https://github.com/code-dot-org/code-dot-org/pull/55363

**Jira ticket:** [A11Y-49](https://codedotorg.atlassian.net/browse/A11Y-49)

----

### Stats carousel
| Before | After |
| ------ | ------ | 
| Promote before | Promote after | 

https://github.com/code-dot-org/code-dot-org/assets/9256643/e996f813-01a4-4688-aa88-56ed3d523935

### Video lightbox
 https://github.com/code-dot-org/code-dot-org/assets/9256643/092dda71-d799-4d8b-b92f-5f916c40cc72

### CODE REVIEW

File: pegasus/sites.v3/code.org/views/share_buttons.haml [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
+      %a.link-button.secondary.small.window-popup.share-button-facebook{href: "https://www.facebook.com/sharer/sharer.php?#{facebook.to_query}"}
+        %i{class: "fa-brands fa-facebook"}
+        =I18n.t(:share_on_facebook)
     -if twitter
-      %a.window-popup.share-button-twitter-link{:style=>"display: none", :href=>"https://twitter.com/intent/tweet?#{twitter.to_query}"}<
-        %button.button-color-share.share-button-twitter<
-          %i{:class=>"fa fa-twitter"}<
-          =I18n.t(:share_on_twitter)
-    -download_url ||= nil
-    -unless download_url.nil_or_empty?
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/views/share_buttons.haml:27]: This doesn't appear to be used anywhere this view is used so I removed it.

File: pegasus/sites.v3/code.org/views/stats_carousel.haml [hunk 2]
```diff
@@ -18,15 +18,13 @@
...
-    %button.button-color-share<
-      More Stats
+  %a.link-button.secondary.small{href: "/files/computer_science_advocacy.pptx"}
+    %i{class: "fa fa-download"}
+    Download presentation
+  %a.link-button.secondary.small{href: "https://code.org/promote/morestats"}
+    More Stats
 .stats-sources-link
-  %a{:href=>'https://docs.google.com/document/d/1gySkItxiJn_vwb8HIIKNXqen184mRtzDX12cux0ZgZk/pub'}<
+  %a{href: "https://docs.google.com/document/d/1gySkItxiJn_vwb8HIIKNXqen184mRtzDX12cux0ZgZk/pub"}
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/views/stats_carousel.haml:27]: Refactored this haml code to use modern markup.

File: pegasus/sites.v3/code.org/views/stats_mobile.haml [hunk 3]
```diff
@@ -3,11 +3,9 @@
...
-  %a{:href=>'/files/computer_science_advocacy.pptx'}
-    %button.button-color-share{:style => 'float: left;'}<
-      %i{:class=>"fa fa-download"}<
-      Download presentation
+  %a.link-button.secondary.small{href: "/files/computer_science_advocacy.pptx"}
+    %i{class: "fa fa-download"}
+    Download presentation
 .stats-sources-link
-  %a{:href=>'https://docs.google.com/document/d/1gySkItxiJn_vwb8HIIKNXqen184mRtzDX12cux0ZgZk/pub'}<
+  %a{href: "https://docs.google.com/document/d/1gySkItxiJn_vwb8HIIKNXqen184mRtzDX12cux0ZgZk/pub"}
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/views/stats_mobile.haml:10]: Refactored this haml code to use modern markup.

File: pegasus/sites.v3/code.org/views/share_buttons.haml [hunk 4]
```diff
@@ -1,29 +1,31 @@
...
+
+      & a {
+        margin: 0;
+      }
+    }
+  }
+
 - unless user_type == "student_y"
-  .share-buttons.clear
-    -if gplus
```
Comment (kelbyhawn) [inline pegasus/sites.v3/code.org/views/share_buttons.haml:11]: Google+ doesn't exist anymore!

**Review by TurnerRiley — APPROVED** [review]
Looks great! Thank you for all the extra cleanup too!

**Pull request merged**